proxy/src/coco/git_helper.rs

```diff
@@ -0,0 +1,46 @@
+//! git-remote-rad git helper related functionality.
+
+use crate::config;
+
+use crate::error;
+use std::fs;
+use std::os::unix::fs::PermissionsExt;
+
+/// Filename of the git helper binary.
+pub const GIT_REMOTE_RAD: &str = "git-remote-rad";
+
+/// Check if a path contains an executable file.
+fn is_executable(path: &std::path::PathBuf) -> Result<bool, error::Error> {
+    let metadata = path.metadata()?;
+    let permissions = metadata.permissions();
+    Ok(permissions.mode() & 0o111 != 0)
+}
+
+/// Checks if the git-remote-rad helper is in a stable location and has the
+/// executable flag, if not copies the executable to the right place.
+///
+/// # Errors
+///
+///   * Could not get the path to directory where helper binaries should be stored.
+///   * Could not get the current working directory.
+///   * Could not create the path to binary directory.
+///   * Could not copy helper executable to the binary directory.
+pub fn setup() -> Result<(), error::Error> {
+    log::info!("Making sure git-remote-rad helper is set up");
+
+    let bin_dir = config::bin_dir()?;
+    let full_dest_path = bin_dir.join(GIT_REMOTE_RAD);
+
+    if full_dest_path.exists() && is_executable(&full_dest_path)? {
```

Comment:
Should we beyond existance check also validate that it is the binary we expect?

Reply:
I think for the first iteration it is sufficient to just make sure it's there and executable.